From: "Barak A. Pearlmutter" <bap@debian.org>
Date: Mon, 24 Jun 2013 18:05:04 +0100
Subject: fix-snefru-segfault

Via: git://src.fedoraproject.org/mhash.git
Taken from Gentoo:
http://mirror.its.uidaho.edu/pub/gentoo-portage/app-crypt/mhash/files/mhash-0.9.9-fix-snefru-segfault.patch
---
 lib/snefru.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/snefru.c b/lib/snefru.c
index 6c10e22..7cfe745 100644
--- a/lib/snefru.c
+++ b/lib/snefru.c
@@ -859,6 +859,8 @@ static void snefru_digest(__const struct snefru_ctx *ctx, mutils_word8 *digest,
 {
 	mutils_word32 i;
 
+	if(!digest) return;
+
 	for (i = 0; i < len; i++, digest += 4)
 	{
 		*(mutils_word32 *)digest = mutils_bend2sys32(ctx->hash[i]);
